File: lib/src/main/java/com/github/goto1134/structurizr/export/d2/model/D2Direction.java
package com.github.goto1134.structurizr.export.d2.model;

/**
 * The direction of the diagram elements arrangement
 *
 * @see <a href="https://d2lang.com/tour/layouts#direction">D2 Direction</a>
 */
public enum D2Direction {
    BOTTOM_TO_TOP("up"),
    TOP_TO_BOTTOM("down"),
    RIGHT_TO_LEFT("left"),
    LEFT_TO_RIGHT("right");
    private final String name;

    D2Direction(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: lib/src/main/kotlin/io/github/goto1134/structurizr/export/d2/model/D2Direction.kt
package io.github.goto1134.structurizr.export.d2.model

/**
 * The direction of the diagram elements arrangement
 *
 * @see [D2 Direction](https://d2lang.com/tour/layouts/#direction)
 */
enum class D2Direction(private val value: String) {
    BOTTOM_TO_TOP("up"),
    TOP_TO_BOTTOM("down"),
    RIGHT_TO_LEFT("left"),
    LEFT_TO_RIGHT("right");

    override fun toString() = value
}
